### PR TITLE
fix(clippy): Resolve some lifetime and reference lints 

### PR DESCRIPTION
**Pull request by teor2345**
## Motivation

Clippy has some new lints on nightly, we should resolve them to avoid subtle bugs.

## Solution

- Fix significant drop in match scrutinee by copying (small) values
- Fix deref immutable value
- Fix `get(0)` when `first()` works and is clearer

## Review

Anyone can review this PR, it is optional.

### Reviewer Checklist

  - [ ] Existing tests pass

### CODE REVIEW

File: zebra-chain/src/transparent/serialize.rs [hunk 1]
```diff
@@ -98,7 +98,7 @@ pub(crate) fn parse_coinbase_height(
     mut data: Vec<u8>,
 ) -> Result<(block::Height, CoinbaseData), SerializationError> {
     use block::Height;
-    match (data.get(0), data.len()) {
+    match (data.first(), data.len()) {
```
Comment (dconnolly) [inline zebra-chain/src/transparent/serialize.rs:101]: ♥️

**Review by dconnolly — APPROVED** [review]
📎🖇️🦀

**Review by upbqdn — APPROVED** [review]

**Comment by upbqdn** [comment]
I was wondering about the `significant_drop_in_scrutinee` lint, and here's a good explanation https://rust-lang.github.io/rust-clippy/master/index.html#significant_drop_in_scrutinee.

**Comment by teor2345** [comment]
@mergifyio update

**Pull request merged**